src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp

```diff
@@ -213,18 +214,32 @@ void S3ObjectStorage::listPrefix(const std::string & path, BlobsPathToSize & chi
 void S3ObjectStorage::removeObject(const std::string & path)
 {
     auto client_ptr = client.get();
-    Aws::S3::Model::ObjectIdentifier obj;
-    obj.SetKey(path);
+    auto settings_ptr = s3_settings.get();
 
-    Aws::S3::Model::Delete delkeys;
-    delkeys.SetObjects({obj});
+    // If chunk size is 0, only use single delete request
+    // This allows us to work with GCS, which doesn't support DeleteObjects
+    if (settings_ptr->objects_chunk_size_to_delete == 0)
```

Comment:
Maybe better to add a separate setting?

Reply:
Thanks for the feedback! Yeah, can go ahead and add another setting.